tracer/src/Datadog.Trace/Headers/Ip/IpExtractor.cs

```diff
@@ -124,7 +122,7 @@ internal static bool IsPrivateIp(IPAddress ipAddress)
 #if NET6_0_OR_GREATER
             return ipAddress.IsIPv6UniqueLocal;
 #else
-            var firstWord = ipAddress.ToString().Split(new[] { ':' }, StringSplitOptions.RemoveEmptyEntries)[0];
+            var firstWord = ipAddress.ToString().Split([':'], StringSplitOptions.RemoveEmptyEntries)[0];
```

Comment:
NIT: This way we avoid creating an array.

```suggestion
            var firstWord = ipAddress.ToString().Split(':', StringSplitOptions.RemoveEmptyEntries)[0];
```

Reply:
Actually, I was checking that, in net9, internally, the library creates a ReadOnlySpan<char> when we send a single char... 

public string[] Split(char separator, StringSplitOptions options = StringSplitOptions.None)
{
    return SplitInternal(new ReadOnlySpan<char>(in separator), int.MaxValue, options);
}

Reply:
this is not compatible with .net461 unfortunately

Reply:
For general string slitting, we have an extension method `string.SplitIntoSpans()` to help split strings without allocating the string array and all the individual strings. You can find an example usage in `W3CBaggagePropagator.ParseHeader()` here:
https://github.com/DataDog/dd-trace-dotnet/blob/a1dee5693ec32fca18e17da83c1e2a2b579a811f/tracer/src/Datadog.Trace/Propagators/W3CBaggagePropagator.cs#L250

But if you only need the string before the first `':'`, you can use `IndexOf(':')` and `Substring()`.

Reply:
thanks for the valuable info! as we only need the first word I used substring in [0433fa3](https://github.com/DataDog/dd-trace-dotnet/pull/6591/commits/0433fa354333ba409777d11984cfb2836c7165f6)

Reply:
Thanks!